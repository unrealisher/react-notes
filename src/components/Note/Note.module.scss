.note {
  display: flex;
  flex-direction: column;
  height: 100%;
  background-color: white;
  border-radius: 16px;
  overflow: hidden;
  box-sizing: border-box;
  border: 1px solid rgba(218, 220, 224, 0.5);
}
