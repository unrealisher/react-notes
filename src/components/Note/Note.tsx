import React from "react";

import INote from "../../interfaces/INote";

import Reminder from "./../Reminder/Reminder";
import NoteList from "./../NoteList/NoteList";
import NoteText from "./../NoteText/NoteText";
import NoteImage from "./../NoteImage/NoteImage";
import Attachments from "./../Attachments/Attachments";

import styles from "./Note.module.scss";

interface IProps {
  note?: INote;
  color?: string;
  tags?: string[] | undefined;
}

const Note = (props: IProps): JSX.Element => {
  const { note, color, tags }: IProps = props;

  return (
    <div className={styles.note}>
      {note && note.reminder && <Reminder reminder={note.reminder} />}
      {note && note.type === "list" && (
        <NoteList note={note} color={color} tags={tags} />
      )}
      {note && note.type === "text" && (
        <NoteText note={note} color={color} tags={tags} />
      )}
      {note && note.type === "image" && <NoteImage note={note} tags={tags} />}
      {note && note.attachments && (
        <Attachments attachments={note.attachments} />
      )}
    </div>
  );
};

export default Note;
