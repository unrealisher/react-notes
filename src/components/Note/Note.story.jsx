import React from "react";

import Note from "./Note";

import { storiesOf } from "@storybook/react";

import "./../../index.scss";

const note = {
  type: "list",
  title: "Список покупок",
  tags: [0],
  color: 0,
  item: [
    {
      text: "Оливочки",
      checked: false
    },
    {
      text: "Макароны",
      checked: false
    },
    {
      text: "Яйца",
      checked: true
    },
    {
      text: "Сливки",
      checked: false
    },
    {
      text: "Хлеб",
      checked: true
    },
    {
      text: "Чеснок",
      checked: false
    },
    {
      text: "Сыр",
      checked: true
    },
    {
      text: "Помидорки",
      checked: false
    },
    {
      text: "Ветчина",
      checked: true
    },
    {
      text: "Чай",
      checked: true
    },
    {
      text: "Огурец",
      checked: false
    },
    {
      text: "Масло сливочное",
      checked: true
    },
    {
      text: "Вино",
      checked: true
    },
    {
      text: "Камамбер",
      checked: true
    },
    {
      text: "Сыр фета",
      checked: true
    },
    {
      text: "Васаби",
      checked: true
    },
    {
      text: "Соевый соус",
      checked: true
    },
    {
      text: "Вкусняшки",
      checked: false
    },
    {
      text: "Красный имбирь",
      checked: true
    },
    {
      text: "Гвоздика",
      checked: true
    },
    {
      text: "Кардамон",
      checked: true
    },
    {
      text: "Овсяные хлопья",
      checked: true
    },
    {
      text: "Пельмени",
      checked: true
    },
    {
      text: "Сметана",
      checked: true
    }
  ],
  size: "l",
  created: 1551780000000
};

const color = "#E84747";

const tags = ["покупки"];

storiesOf("Note", module).add("list", () => (
  <Note note={note} color={color} tags={tags} />
));
