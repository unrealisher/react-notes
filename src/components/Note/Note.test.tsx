import React from "react";
import { shallow } from "enzyme";
import { Note } from "./Note";

describe("Note", () => {
  const note = {
    id: 1,
    type: "text",
    title: "Завтра дедлайн",
    text: "Не забыть сверстать макет",
    tags: [1, 2],
    color: 2,
    size: "s",
    created: 1551714600000
  };
  const color = "#F2C94C";
  const tags = ["Работа", "ШРИ"];
  const setPatchItem = (): void => {};
  const setPopup = (): void => {};
  it("correct render", () => {
    const component = shallow(
      <Note
        note={note}
        color={color}
        tags={tags}
        setPatchItem={setPatchItem}
        setPopup={setPopup}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
