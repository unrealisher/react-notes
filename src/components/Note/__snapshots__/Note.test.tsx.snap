// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note correct render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Note
    color="#F2C94C"
    note={
      Object {
        "color": 2,
        "created": 1551714600000,
        "id": 1,
        "size": "s",
        "tags": Array [
          1,
          2,
        ],
        "text": "Не забыть сверстать макет",
        "title": "Завтра дедлайн",
        "type": "text",
      }
    }
    setPatchItem={[Function]}
    setPopup={[Function]}
    tags={
      Array [
        "Работа",
        "ШРИ",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        false,
        <NoteText
          color="#F2C94C"
          note={
            Object {
              "color": 2,
              "created": 1551714600000,
              "id": 1,
              "size": "s",
              "tags": Array [
                1,
                2,
              ],
              "text": "Не забыть сверстать макет",
              "title": "Завтра дедлайн",
              "type": "text",
            }
          }
          setPatchItem={[Function]}
          setPopup={[Function]}
          tags={
            Array [
              "Работа",
              "ШРИ",
            ]
          }
        />,
        false,
        undefined,
      ],
      "className": "note",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "color": "#F2C94C",
          "note": Object {
            "color": 2,
            "created": 1551714600000,
            "id": 1,
            "size": "s",
            "tags": Array [
              1,
              2,
            ],
            "text": "Не забыть сверстать макет",
            "title": "Завтра дедлайн",
            "type": "text",
          },
          "setPatchItem": [Function],
          "setPopup": [Function],
          "tags": Array [
            "Работа",
            "ШРИ",
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          false,
          <NoteText
            color="#F2C94C"
            note={
              Object {
                "color": 2,
                "created": 1551714600000,
                "id": 1,
                "size": "s",
                "tags": Array [
                  1,
                  2,
                ],
                "text": "Не забыть сверстать макет",
                "title": "Завтра дедлайн",
                "type": "text",
              }
            }
            setPatchItem={[Function]}
            setPopup={[Function]}
            tags={
              Array [
                "Работа",
                "ШРИ",
              ]
            }
          />,
          false,
          undefined,
        ],
        "className": "note",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "color": "#F2C94C",
            "note": Object {
              "color": 2,
              "created": 1551714600000,
              "id": 1,
              "size": "s",
              "tags": Array [
                1,
                2,
              ],
              "text": "Не забыть сверстать макет",
              "title": "Завтра дедлайн",
              "type": "text",
            },
            "setPatchItem": [Function],
            "setPopup": [Function],
            "tags": Array [
              "Работа",
              "ШРИ",
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
